independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/core/Serialisers.java

```diff
@@ -76,18 +76,24 @@ public List<MessageBodyReader<?>> findReaders(ConfigurationImpl configuration, C
                 }
             }
             List<ResourceReader> goodTypeReaders = readers.get(klass);
-            readerLookup(mediaType, runtimeType, mt, ret, goodTypeReaders);
+            readerLookup(mediaType, runtimeType, desired, ret, goodTypeReaders);
             if (klass.isInterface()) {
                 klass = Object.class;
             } else {
                 klass = klass.getSuperclass();
             }
         } while (klass != null);
 
-        return ret;
+        List<MessageBodyReader<?>> matchingReaders = new ArrayList<>();
+        for (ResourceReader mediaTypeMatchingReader : ret) {
```

Comment:
Can we avoid the use of the iterator here? We do that in various parts of the code in order to cut down on the number of objects created while serving a request

Reply:
Done. To be precise, I reverted this change.